### PR TITLE
Allow rgbasm and rgblink to use stdout and stdin as input and output

### PR DESCRIPTION
**Pull request by daid**
rgbfix needs quite some rewrite to enable this, so that is for a later time.

### CODE REVIEW

File: src/link/main.c [hunk 1]
```diff
@@ -126,7 +127,14 @@ FILE *openFile(char const *fileName, char const *mode)
 	if (!fileName)
 		return NULL;
 
-	FILE *file = fopen(fileName, mode);
+	FILE *file;
+	if (strcmp(fileName, "-") != 0) {
+		file = fopen(fileName, mode);
+	} else if (strchr(mode, 'r')) {
```
Comment (ISSOtm) [inline src/link/main.c:133]: Given that `r` has to be the first character, it'd be better to just check that.

Also, in theory, the braces don't match our code standard, but I guess a `clang-format` *really* is needed...

**Review by ISSOtm — CHANGES_REQUESTED** [review]

File: src/link/main.c (hunk 1 above)
Comment (daid) [reply src/link/main.c:133]: I'm not seeing the code style mismatch? The other point you are right, didn't know that.

Reference: https://github.com/gbdev/rgbds/blob/master/src/asm/lexer.c#L977

Comment (ISSOtm) [reply src/link/main.c:133]: Since our style is a mess anyways (#612), I don't really care.

Comment (AntonioND) [reply src/link/main.c:133]: The braces/no braces thing is pretty annoying in the case of the Linux style. Having braces is usually better, as it prevents problems like: https://blog.codecentric.de/en/2014/02/curly-braces/

Comment (daid) [reply src/link/main.c:133]: Oh that way having braces for single statements. I usually put them in as soon as I have an else.

But this was also a case where I initially had some logging lines in there to make sure it worked.

**Comment by ISSOtm** [comment]
macOS fails for some reason, but [it's already the case on `master`](https://github.com/gbdev/rgbds/runs/1350730239?check_suite_focus=true#step:4:92)... Merging.

**Pull request merged**